common/transport/mqtt/src/mqtt_base.ts

```diff
@@ -108,7 +108,7 @@ export class MqttBase extends EventEmitter {
             /*Codes_SRS_NODE_COMMON_MQTT_BASE_16_033: [The `updateSharedAccessSignature` method shall disconnect and reconnect the mqtt client with the new `sharedAccessSignature`.]*/
             /*Codes_SRS_NODE_COMMON_MQTT_BASE_16_035: [The `updateSharedAccessSignature` method shall call the `callback` argument with no parameters if the operation succeeds.]*/
             /*Codes_SRS_NODE_COMMON_MQTT_BASE_16_036: [The `updateSharedAccessSignature` method shall call the `callback` argument with an `Error` if the operation fails.]*/
-            this._disconnectClient(false, () => {
+            this._disconnectClient(true, () => {
```

Comment:
>true, [](start = 35, length = 5)

There's a very weak connection between this change and the effect it has.  Is there a better way to do this?  "forceClose" always feels like a blunt tool, and I've mostly seen it used when shutting down or dealing with errors.  This seems like a non-error case that needs to be forced, but I'm not sure why.

Reply:
unfortunately we have I don't know of any way to detect if a network connection is up or not. I guess we could try to disconnect properly with a timeout and past that timeout, force the disconnect.. and tbh i'm not sure what the downside of forcing the disconnect is.

Reply:
Turns out there is a downside, which is to screw up the existing subscriptions, potentially messing features such as device methods after an update to the shared access signature has happened. I'm reverting this for now until we find a better solution. this problem is separate from the premature close error anyways...